include/realizations/catchment/Bmi_Module_Formulation.hpp

```diff
@@ -419,6 +426,7 @@ namespace realization {
         int next_time_step_index = 0;
 
     private:
+        models::bmi::protocols::NgenBmiProtocols bmi_protocols;
```

Comment:
It should be easy to change in the future. But, I can't get it out of my head wanting to associate a single instance of a `NgenBmiProtocol` with multiple `Formulation` instances. I think expanding on what you have with `models::bmi::protocols::NgenBmiProtocols` will make that possible. Im just leaving my thoughts along the way :).

Reply:
Ah I remember my thought. This relates to what we talked about below when differentiating between "stateless" and "stateful" protocols instances. My thinking is there is no reason for multiple instances of a "stateless" protocol. We should be able to reuse that instance for all formulation instances. I think we can shove this down into the implementation of `NgenBmiProtocols` if we ever go that route.